integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/swagger/SwaggerGenerationTestCase.java

```diff
@@ -0,0 +1,132 @@
+            "{\"type\":\"object\",\"properties\":{\"payload\":{\"type\":\"object\",\"properties\":{\"productLine\":" +
+            "{\"type\":\"string\"},\"quantityInStock\":{\"format\":\"int32\",\"type\":\"integer\"},\"buyPrice\":" +
+            "{\"format\":\"double\",\"type\":\"number\"},\"productCode\":{\"type\":\"string\"},\"productName\":" +
+            "{\"type\":\"string\"}}}}},\"in\":\"body\",\"name\":\"payload\",\"description\":\"Sample Payload\"," +
+            "\"required\":false}]},\"put\":{\"responses\":{\"default\":{\"description\":\"this is the default " +
+            "response\"}},\"parameters\":[{\"schema\":{\"type\":\"object\",\"properties\":{\"payload\":{\"type\"" +
+            ":\"object\",\"properties\":{\"productLine\":{\"type\":\"string\"},\"quantityInStock\":{\"format\":" +
+            "\"int32\",\"type\":\"integer\"},\"buyPrice\":{\"format\":\"double\",\"type\":\"number\"},\"productCode\"" +
+            ":{\"type\":\"string\"},\"productName\":{\"type\":\"string\"}}}}},\"in\":\"body\",\"name\":\"payload\"," +
+            "\"description\":\"Sample Payload\",\"required\":false}]}},\"/product/{productCode}\":{\"get\"" +
+            ":{\"responses\":{\"default\":{\"description\":\"this is the default response\"}},\"parameters\":" +
+            "[{\"in\":\"path\",\"name\":\"productCode\",\"type\":\"string\",\"required\":true}]},\"delete\":" +
+            "{\"responses\":{\"default\":{\"description\":\"this is the default response\"}},\"parameters\":" +
+            "[{\"in\":\"path\",\"name\":\"productCode\",\"type\":\"string\",\"required\":true},{\"schema\":" +
+            "{\"type\":\"object\",\"properties\":{\"payload\":{\"type\":\"object\",\"properties\":{\"productCode\":" +
+            "{\"type\":\"string\"}}}}},\"in\":\"body\",\"name\":\"payload\",\"description\":\"Sample Payload\"," +
+            "\"required\":false}]}},\"/getProductWithCode/{productCode}/andName/{productName}\":{\"get\":" +
+            "{\"responses\":{\"default\":{\"description\":\"this is the default response\"}},\"parameters\":" +
+            "[{\"in\":\"path\",\"name\":\"productCode\",\"type\":\"string\",\"required\":true},{\"in\":\"path\"," +
+            "\"name\":\"productName\",\"type\":\"string\",\"required\":true}]}},\"/products\":{\"get\":{\"responses\":" +
+            "{\"default\":{\"description\":\"this is the default response\"}},\"parameters\":[]}}}";
+
+    @BeforeClass(alwaysRun = true)
+    public void serviceDeployment() throws Exception {
+        super.init();
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void destroy() throws Exception {
+        cleanup();
+    }
+
+    @Test(groups = {"wso2.dss"}, description = "Check swagger generation feature")
+    public void swaggerDataServiceWithResourcesTestCase() throws Exception {
+
+        String serviceEndPoint = "http://localhost:8480/services/ResourcesServiceTest?swagger.json";
+        HttpResponse response = this.getHttpResponse(serviceEndPoint);
```

Comment:
Shall we simple use HttpRequestUtil.doPost(new URL(url), PAYLOAD, headers); .. we don't need a method to get response as well.

Reply:
+1 fixed